Upgrade FluxC to Android 11

ravishanker opened the pull request:
Fixes [WordPress-Android#15338](https://github.com/wordpress-mobile/WordPress-Android/issues/15338)

Upgrade to Android 11 (API level 30)

Please see paqN3M-mr-p2 for details on mandatory requirement to upgrade to Android 11 by November 2021.

- upgrade minSdkVersion to 24 and targetSdkVersion, compileSdkVersion to 30

ravishanker committed:
Upgrade to Android 11

Upgrade to Android 11 (API level 30)

ravishanker commented on fluxc/build.gradle:
```diff
@@ -9,11 +9,11 @@ plugins {
+    compileSdkVersion 30
 
     defaultConfig {
-        minSdkVersion 18
```
Is there a reason why `minSdkVersion` is still at 18 here?

kidinov replied:
I would ask it from the oposite side - is there a reason why we need to bump it? =)

malinajirka replied:
⚠️ Just a headsup that WooAndroid still uses api 21 -> so if we want to bump it, we might want to consider bumping it only to 21 so detekt/lint/AS keeps notifying us about issues. Wdyt?

ravishanker replied:
Reverted to 18.  Looks like if it is upgraded then all the others like instaflux, plugins.woocommerce..

ravishanker committed:
Update build.gradle

Revert minSdkVersion to 18

ravishanker committed:
Upgrade instaflux and woo to Android 11

Upgrade instaflux and plugin.woocommerce

ravishanker committed:
Upgrade example

Upgrade example app to 30, and make some necessary changes to make work

ravishanker committed:
fix lint issues

malinajirka commented on example/src/main/java/org/wordpress/android/fluxc/example/SiteSelectorDialog.kt:
```diff
@@ -76,7 +76,7 @@ class SiteSelectorDialog : DialogFragment() {
                         val adapter = (dialog as AlertDialog).listView.adapter as SiteAdapter
                         val site = adapter.getItem(which)
-                        listener?.onSiteSelected(site, which)
+                        site?.let { siteModel -> listener?.onSiteSelected(siteModel, which) }
```
💡 We might want to consider adding something like `prependToLog("SiteChanged error: site at position $which was null.")`. I personally wouldn't even mind crashing since this is just an example app and it's better to know about any issue. Wdyt?

ravishanker replied:
- [x] done

malinajirka commented on example/src/main/java/org/wordpress/android/fluxc/example/ui/StoreSelectorDialog.kt:
```diff
@@ -50,7 +50,7 @@ class StoreSelectorDialog : DialogFragment() {
                         val adapter = (dialog as AlertDialog).listView.adapter as SiteAdapter
                         val site = adapter.getItem(which)
-                        listener?.onSiteSelected(site, which)
+                        site?.let { siteModel -> listener?.onSiteSelected(siteModel, which) }
```
Same as above

ravishanker replied:
- [x] done

malinajirka commented on example/src/main/java/org/wordpress/android/fluxc/example/ui/customer/search/WooCustomersSearchFragment.kt:
```diff
@@ -24,15 +24,15 @@ class WooCustomersSearchFragment : Fragment() {
-                customerRole = searchParams.role,
-                customerRemoteCustomerIds = searchParams.includeIds,
-                customerExcludedCustomerIds = searchParams.excludeIds
+                customerSearchQuery = searchParams?.searchQuery,
```
Same as above

ravishanker replied:
- [x] done

malinajirka reviewed (commented):
Thanks for working on this @ravishanker! I've left three minor non-blocking comments - all about logging an error or following the fail fast principle. 

I'll approve the PR when the build is fixed and I can smoke-test the changes in WooAndroid.

ravishanker committed:
Remove null safety check

Removed null safety checks and added logging to identify the issue by letting it crash since it is an example app

malinajirka reviewed (approved):
LGTM, thanks for the changes ;)! 

Feel free to merge this. I'm not merging it since I'm not sure if it's waiting for some other PRs or should be merged right away.

ravishanker committed:
Merge branch 'develop' into fix/15322-upgrade-to-android-11-api-level-30
